src/sources/socket/udp.rs

```diff
@@ -98,17 +100,20 @@ pub fn udp(
                         })
                     })?;
 
+                    emit!(&BytesReceived { byte_size, protocol: "udp" });
+
                     let payload = buf.split_to(byte_size);
 
                     let mut stream = FramedRead::new(payload.as_ref(), decoder.clone());
 
-                    loop {
-                        match stream.next().await {
-                            Some(Ok((mut events, byte_size))) => {
+                    while let Some(result) = stream.next().await {
+                        match result {
+                            Ok((mut events, byte_size)) => {
+                                let count = events.len();
                                 emit!(&SocketEventsReceived {
                                     mode: SocketMode::Udp,
                                     byte_size,
```

Comment:
Similar to my comments on other PRs, the byte sized used for `component_received_event_bytes_total` should be the sum of in-memory bytes calculated via `ByteSizeOf::size_of`.